spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ReactiveKafkaAutoConfiguration.java

```diff
@@ -0,0 +1,118 @@
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.context.properties.PropertyMapper;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for the Reactive client of Apache
+ * Kafka.
+ *
+ * @author Almog Tavor
+ * @since 2.7.0
+ */
+@AutoConfiguration
+@ConditionalOnClass({ KafkaReceiver.class, KafkaSender.class })
+@ConditionalOnBean(KafkaProperties.class)
+@EnableConfigurationProperties(ReactiveKafkaProperties.class)
+public class ReactiveKafkaAutoConfiguration {
+
+	private final KafkaProperties kafkaProperties;
+
+	private final ReactiveKafkaProperties reactiveKafkaProperties;
+
+	private static final PropertyMapper map = PropertyMapper.get();
+
+	public ReactiveKafkaAutoConfiguration(KafkaProperties kafkaProperties,
+			ReactiveKafkaProperties reactiveKafkaProperties) {
+		this.kafkaProperties = kafkaProperties;
+		this.reactiveKafkaProperties = reactiveKafkaProperties;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(ReceiverOptions.class)
+	public <K, V> ReceiverOptions<K, V> receiverOptions() {
+		Map<String, Object> properties = this.kafkaProperties.buildConsumerProperties();
+		ReceiverOptions<K, V> receiverOptions = ReceiverOptions.create(properties);
+		ReactiveKafkaProperties.Receiver receiverProperties = this.reactiveKafkaProperties.getReceiver();
+		receiverOptions = setPropertyWhenGreaterThanZero(receiverProperties.getAtmostOnceCommitAheadSize(),
+				receiverOptions::atmostOnceCommitAheadSize, receiverOptions);
+		receiverOptions = setPropertyWhenGreaterThanZero(receiverProperties.getMaxDeferredCommits(),
+				receiverOptions::maxDeferredCommits, receiverOptions);
+		receiverOptions = setPropertyWhenGreaterThanZero(receiverProperties.getMaxCommitAttempts(),
+				receiverOptions::maxCommitAttempts, receiverOptions);
+		receiverOptions = setPropertyWhenGreaterThanZero(receiverProperties.getCommitBatchSize(),
+				receiverOptions::commitBatchSize, receiverOptions);
+		receiverOptions = setPropertyWhenNonNull(receiverProperties.getCloseTimeout(), receiverOptions::closeTimeout,
+				receiverOptions);
+		receiverOptions = setPropertyWhenNonNull(receiverProperties.getPollTimeout(), receiverOptions::pollTimeout,
+				receiverOptions);
+		receiverOptions = setPropertyWhenNonNull(receiverProperties.getCommitInterval(),
+				receiverOptions::commitInterval, receiverOptions);
+		receiverOptions = setPropertyWhenNonNull(receiverProperties.getSubscribeTopics(), receiverOptions::subscription,
+				receiverOptions);
+		if (Optional.ofNullable(receiverProperties.getSubscribeTopics()).isEmpty()) {
+			receiverOptions = setPropertyWhenNonNull(receiverProperties.getSubscribePattern(),
+					receiverOptions::subscription, receiverOptions);
+		}
+		return receiverOptions;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(SenderOptions.class)
```

Comment:
The bean type is redundant here as the return type of the method is the default.